### PR TITLE
feat(memory): add E2E memory persistence smoke test (Issue #47)

### LINKED ISSUE #47: Add end-to-end memory smoke test for persistence and isolation
## Context
The most important product behavior is that Sara remembers John across sessions, while Luna does not receive Sara's memories. This should be covered by an integration-style smoke test using the FastAPI websocket path.

## Scope
Add a smoke or integration test that exercises memory across chat turns and companion boundaries. Use stubs/mocks where needed so CI does not require a real GPU model.

Suggested location:
- `tests/test_memory_chat_flow.py` or extend `scripts/smoke_stack.py` carefully

Scenario:
1. Create user John.
2. Create user companion preferences.
3. Create AI companions Sara and Luna.
4. Send Sara a message containing an explicit preference to remember.
5. Allow/trigger extraction to persist memory.
6. Reconnect or start another chat request with Sara and verify retrieved memory is injected.
7. Send a chat request to Luna and verify Sara's memory is not injected.

Use deterministic runtime/extraction/vector stubs so tests are stable.

## Acceptance Criteria
- Test proves memory survives beyond the rolling 24-message context path.
- Test proves `user_id + ai_companion_id` isolation.
- Test does not require external Qdrant or model downloads in normal CI.
- Test fails if retrieved memories are inserted as visible chat messages instead of system prompt context.
- Existing tests remain passing.

### PR DESCRIPTION
**Pull request by umang-910**
This PR adds a comprehensive integration smoke test for the memory subsystem, verifying extraction, persistence, and isolation. It also updates the Qdrant image tag to v1.17.1 to support the query_points API.

Closes #47

### CODE REVIEW

File: tests/unit/test_memory_chat_flow.py [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+        while True:
+            resp = ws.receive_json()
+            if resp["type"] == "done":
+                break
+                
+    # Check that system prompt injected the memory
+    assert len(stream_calls) == 1
+    sara_prompt = stream_calls[0]
+    assert "User loves skydiving" in sara_prompt
+    assert "LONG-TERM MEMORY (CURATED)" in sara_prompt
```
Comment (vatsal1306) [inline tests/unit/test_memory_chat_flow.py:350]: This test reconnects to Sara immediately after the first message, so the original `"I love skydiving"` message is still inside normal recent history. Add enough filler turns/messages after extraction to push the original preference outside `history_message_limit`, then verify the memory is still injected into the system prompt.

**Review by vatsal1306 — CHANGES_REQUESTED** [review]

**Review by vatsal1306 — APPROVED** [review]
